crates/router/src/connector/mifinity/transformers.rs

```diff
@@ -48,67 +109,160 @@ impl TryFrom<&MifinityRouterData<&types::PaymentsAuthorizeRouterData>> for Mifin
         item: &MifinityRouterData<&types::PaymentsAuthorizeRouterData>,
     ) -> Result<Self, Self::Error> {
         match item.router_data.request.payment_method_data.clone() {
-            domain::PaymentMethodData::Card(req_card) => {
-                let card = MifinityCard {
-                    number: req_card.card_number,
-                    expiry_month: req_card.card_exp_month,
-                    expiry_year: req_card.card_exp_year,
-                    cvc: req_card.card_cvc,
-                    complete: item.router_data.request.is_auto_capture()?,
-                };
-                Ok(Self {
-                    amount: item.amount.to_owned(),
-                    card,
-                })
+            domain::PaymentMethodData::Wallet(wallet_data) => match wallet_data {
+                domain::WalletData::Mifinity(data) => {
+                    let money = Money {
+                        amount: item.amount.clone(),
+                        currency: item.router_data.request.currency.to_string(),
+                    };
+                    let phone_details = item.router_data.get_billing_phone()?;
+                    let client = MifinityClient {
+                        first_name: item.router_data.get_billing_first_name()?,
+                        last_name: item.router_data.get_billing_last_name()?,
+                        phone: phone_details.get_number()?,
+                        dialing_code: phone_details.get_country_code()?,
+                        nationality: item.router_data.get_billing_country()?,
+                        email_address: item.router_data.get_billing_email()?,
+                        dob: data.dob.clone(),
+                    };
+                    let address = MifinityAddress {
+                        address_line1: item.router_data.get_billing_line1()?,
+                        country_code: item.router_data.get_billing_country()?,
+                        city: item.router_data.get_billing_city()?,
+                    };
+                    let validation_key = format!(
+                        "payment_validation_key_{}_{}",
+                        item.router_data.merchant_id,
+                        item.router_data.connector_request_reference_id.clone()
+                    );
+                    let client_reference = item.router_data.customer_id.clone().ok_or(
+                        errors::ConnectorError::MissingRequiredField {
+                            field_name: "client_reference",
+                        },
```

Comment:
Can this be our payment_id?

Reply:
Client reference is "Id of client in merchant’s system"